lib/kernel/read_image.cl

```diff
@@ -66,37 +71,32 @@ int __pocl_is_out_of_bounds (ADDRESS_SPACE dev_image_t* dev_image, int4 *coord,
          coord->y >= dev_image->_height || coord->y < 0 ||
          (dev_image->_depth != 0 && (coord->z >= dev_image->_depth || coord->z <0)))
         {
-          (*color)[0] = 0;
-          (*color)[1] = 0;
-          (*color)[2] = 0;
-
           if (dev_image->_order == CL_A || dev_image->_order == CL_INTENSITY ||
               dev_image->_order == CL_RA || dev_image->_order == CL_ARGB ||
               dev_image->_order == CL_BGRA || dev_image->_order == CL_RGBA)
-            (*color)[3] = 0;
+            return 1; // clamp to 0,0,0,0
           else
-            (*color)[3] = 1;
-
-          return 1;
+            return 2; // clamp to 0,0,0,1
         }
     }
   return 0;
 }
 
 /* Reads a four element pixel from image pointed by integer coords. */
-void __pocl_read_pixel (void* color, ADDRESS_SPACE dev_image_t* dev_image, int4 coord)
+static void __pocl_read_pixel (void* color, global dev_image_t* dev_image, int4 coord)
 {
 
   uint4* color_ptr = (uint4*)color;
   int width = dev_image->_width;
```

Comment:
I think the underscore prefixed names are wrong in the struct too, but I understand it's not your change.